[TensorExpr] Properly handle all dtypes in evaluation of CompareSelect exprs.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #42495 [TensorExpr] Properly handle all dtypes of the condition in evaluation of IfThenElse exprs.
* #42494 [TensorExpr] Properly handle all dtypes in evaluation of Intrinsics exprs.
* **#42493 [TensorExpr] Properly handle all dtypes in evaluation of CompareSelect exprs.**

Differential Revision: [D22910754](https://our.internmc.facebook.com/intern/diff/D22910754)

Commit:
[TensorExpr] Properly handle all dtypes in evaluation of CompareSelect exprs.

[ghstack-poisoned]

nickgg commented on test/cpp/tensorexpr/test_expr.cpp:
```diff
@@ -254,6 +254,46 @@ void testExprCompareSelectEQ() {
   assertAllEqual(c_buffer, 1);
 }
 
+void testExprCompareSelectDtypes() {
```
Can we have a comment about this test, as you found out offline I thought you'd implemented something different. It's not obvious that the important part is the difference between the type of the condition and value.

nickgg reviewed (approved):
👍

Commit:
Update on "[TensorExpr] Properly handle all dtypes in evaluation of CompareSelect exprs."


Differential Revision: [D22910754](https://our.internmc.facebook.com/intern/diff/D22910754)

[ghstack-poisoned]
